### PR TITLE
Support kuberflow pytorch-operator

### PR DESCRIPTION
**Pull request by SparkSnail**
1.Support pytorch-operator
2.remove unsupported operator

### CODE REVIEW

File: tools/nni_cmd/launcher_utils.py [hunk 1]
```diff
@@ -130,26 +147,31 @@ def parse_assessor_content(experiment_config):
...
         if experiment_config.get('searchSpacePath'):
-            raise Exception('If you set useAnnotation=true, please leave searchSpacePath empty')
+            print_error('If you set useAnnotation=true, please leave searchSpacePath empty')
+            exit(1)
     else:
         # validate searchSpaceFile
         if experiment_config['tuner'].get('tunerName') and experiment_config['tuner'].get('optimizationMode'):
             if experiment_config.get('searchSpacePath') is None:
-                raise Exception('Please set searchSpace!')
+                print_error('Please set searchSpace!')
```
Comment (chicm-ms) [inline tools/nni_cmd/launcher_utils.py]: 'Please set searchSpace!' , should the searchSpace be searchSpacePath?

Comment (SparkSnail) [reply tools/nni_cmd/launcher_utils.py]: updated to searchSpacePath

File: src/nni_manager/training_service/kubeflow/kubeflowConfig.ts [hunk 2]
```diff
@@ -23,41 +23,64 @@ import { TrialConfig } from "../common/trialConfig";
...
 
 /**
  * Kuberflow cluster configuration
  * 
  */
-export class KubeflowClusterConfig {
+export class KubeflowClusterConfigBase {
     /** Name of Kubeflow operator, like tf-operator */
     public readonly operator: KubeflowOperator;
     public readonly nfs?: NFSConfig;
```
Comment (yds05) [inline src/nni_manager/training_service/kubeflow/kubeflowConfig.ts]: These fields: 
nfs, keyVault, azureStorage
should be removed from base class?

File: src/nni_manager/training_service/kubeflow/kubeflowConfig.ts [hunk 3]
```diff
@@ -142,15 +165,31 @@ export class KubeflowTrialConfigTemplate {
     }
 }
 
-export class KubeflowTrialConfig {
+export class KubeflowTrialConfigBase {
     public readonly codeDir: string;
-    public readonly ps?: KubeflowTrialConfigTemplate;
     public readonly worker: KubeflowTrialConfigTemplate;
```
Comment (yds05) [inline src/nni_manager/training_service/kubeflow/kubeflowConfig.ts]: I think worker can also be moved out from KubeflowTrialConfigBase

File: src/nni_manager/training_service/kubeflow/kubeflowTrainingService.ts [hunk 4]
```diff
@@ -139,22 +138,26 @@ class KubeflowTrainingService implements TrainingService {
 
             await fs.promises.writeFile(path.join(trialLocalTempFolder, 'run_worker.sh'), workerRunScriptContent, { encoding: 'utf8' });
         }
-
         // Write parameter server file content run_ps.sh to local tmp folders
-        if(this.kubeflowTrialConfig.ps) {
+        if(this.kubeflowTrialConfig instanceof KubeflowTrialConfigTensorflow && this.kubeflowTrialConfig.ps) {
```
Comment (yds05) [inline src/nni_manager/training_service/kubeflow/kubeflowTrainingService.ts]: It's good to use this.kubeflowClusterConfig.operator to explicitly convet this.kubeflowTrialConfig to specific sub-class's instance

File: src/nni_manager/training_service/kubeflow/kubeflowTrainingService.ts [hunk 5]
```diff
@@ -164,37 +167,42 @@ class KubeflowTrainingService implements TrainingService {
...
         workerPodResources.limits = Object.assign({}, workerPodResources.requests);
-
-        let psPodResources : any = undefined;
-        if(this.kubeflowTrialConfig.ps) {
-            psPodResources = {};
-            psPodResources.requests = {
+        let psOrMasterPodResources : any = undefined;
+        if(this.kubeflowTrialConfig instanceof KubeflowTrialConfigTensorflow && this.kubeflowTrialConfig.ps) {
+            psOrMasterPodResources = {};
+            psOrMasterPodResources.requests = {
```
Comment (yds05) [inline src/nni_manager/training_service/kubeflow/kubeflowTrainingService.ts]: line 173-178 is dup with line 180-186

File: src/nni_manager/training_service/kubeflow/kubeflowTrainingService.ts [hunk 6]
```diff
@@ -570,15 +622,32 @@ class KubeflowTrainingService implements TrainingService {
         if(this.kubeflowTrialConfig) {
             switch(roleType) {
                 case 'ps':
-                    if(this.kubeflowTrialConfig.ps && this.kubeflowTrialConfig.ps.gpuNum == 0) {
-                        runScriptLines.push(`export CUDA_VISIBLE_DEVICES=''`);
+                    if(this.kubeflowTrialConfig instanceof KubeflowTrialConfigTensorflow){
```
Comment (yds05) [inline src/nni_manager/training_service/kubeflow/kubeflowTrainingService.ts]: suggest to add a param for generateRunScript, like resetGPU : boolean. 
The only thing generateRuScript need to do is:
if(resetGPU) {
   runScriptLines.push(`export CUDA_VISIBLE_DEVICES=''`);
}

File: src/nni_manager/training_service/kubeflow/kubeflowTrainingService.ts [hunk 7]
```diff
@@ -570,15 +622,32 @@ class KubeflowTrainingService implements TrainingService {
...
+                        if(this.kubeflowTrialConfig.ps && this.kubeflowTrialConfig.ps.gpuNum == 0) {
```
Comment (yds05) [inline src/nni_manager/training_service/kubeflow/kubeflowTrainingService.ts]: function ```genereateRunScript``` 's name is typoed. Please help to correct it to ```generateRunScript```

File: src/nni_manager/training_service/kubeflow/kubeflowTrainingService.ts [hunk 8]
```diff
@@ -115,6 +117,15 @@ class KubeflowTrainingService implements TrainingService {
...
             this.kubeflowRestServerPort = restServer.clusterRestServerPort;
         }
+        // initialize kubeflow trial config to specific type
+        let kubeflowTrialConfig;
+        if(this.kubeflowClusterConfig.operator === 'tf-operator') {
+            kubeflowTrialConfig = <KubeflowTrialConfigTensorflow>this.kubeflowTrialConfig;
+        }else if(this.kubeflowClusterConfig.operator === 'pytorch-operator'){
+            kubeflowTrialConfig = <KubeflowTrialConfigPytorch>this.kubeflowTrialConfig;
+        }else {
+            throw Error('kubeflowTrialConfig not initialized!')
```
Comment (yds05) [inline src/nni_manager/training_service/kubeflow/kubeflowTrainingService.ts]: Suggest to change to:
throw Error(`operator ${this.kubeflowClusterConfig.operator} is invalid')

Comment (SparkSnail) [reply src/nni_manager/training_service/kubeflow/kubeflowTrainingService.ts]: update to throw Error(`operator ${this.kubeflowClusterConfig.operator} is invalid')

File: src/nni_manager/training_service/kubeflow/kubeflowTrainingService.ts (hunk 7 above)
Comment (SparkSnail) [reply src/nni_manager/training_service/kubeflow/kubeflowTrainingService.ts]: updated to generateRunScript

**Pull request merged**